### DISCUSSION TITLE
Kafka .withKraft() fails with cp-kafka:latest

### DISCUSSION THREAD

**Discussion by achoice** [testcontainers/testcontainers-java#7966] [Q&A] [answered]
Trying to run Testcontainer Kafka with kraft using **latest kafka**.
Fails due to it requires Zookeeper.
It DOES work when manually specifying latest version`confluentinc/cp-kafka:7.4.3`

```
@Test
void kafkaLatestFail() {
    try(KafkaContainer kafka = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:latest")).withKraft()) {
        kafka.start();
    }
}

/** 7.4.3 IS latest as of today */
@Test
void kafka743ok() {
    try(KafkaContainer kafka = new KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka:7.4.3")).withKraft()) {
        kafka.start();
    }
}
```

**:latest** Fails with this log;
KAFKA_ZOOKEEPER_CONNECT is required.
Command [/usr/local/bin/dub ensure KAFKA_ZOOKEEPER_CONNECT] FAILED !

**Comment by eddumelendez** [comment, marked as answer]
It is recommended to set an specific version. Check the [Testcontainers best practices](https://www.atomicjar.com/2023/11/testcontainers-best-practices/)

**Reply by achoice** [reply]
OH, you are correct - manually specifying version works!
_(thanks for quick help 👍 )_
